internal/utils/ctrlcreader.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020-2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"io"
+	"sync"
+)
+
+// CtrlCReader implements a simple Reader/Closer that returns Ctrl-C and EOF
+// on Read() after it has been closed, and nothing before it.
+type CtrlCReader struct {
+	once sync.Once
+	done chan struct{}
+}
+
+// NewCtrlCReader returns a new CtrlCReader instance
+func NewCtrlCReader() *CtrlCReader {
```

Comment:
Why not make this function take a `context.Context`? Then this doesn't need a `Close()` method, we don't need to bother with the `sync.Once` here, and it saves the caller a lot of busywork to use this correctly.

Reply:
May still need the `Close()` to differentiate between graceful close and forceful cancellation on timeout or (future) early command termination on successful test detection based on flow streaming.

Reply:
Done, but keeping the `Close()` method to allow the reader to be unblocked gracefully without sending Ctrl-C/D.